pkg/reconciler/kubernetes/tektonchains/tektonchains.go

```diff
@@ -0,0 +1,406 @@
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+)
+
+// Reconciler implements controller.Reconciler for TektonChains resources.
+type Reconciler struct {
+	// operatorClientSet allows us to configure operator objects
+	operatorClientSet clientset.Interface
+	// manifest has the source manifest of Tekton Triggers for a
+	// particular version
+	manifest mf.Manifest
+	// Platform-specific behavior to affect the transform
+	// enqueueAfter enqueues a obj after a duration
+	enqueueAfter func(obj interface{}, after time.Duration)
+	extension    common.Extension
+	// chainsVersion describes the current chains version
+	chainsVersion   string
+	operatorVersion string
+	// pipelineInformer provides access to a shared informer and lister for
+	// TektonPipelines
+	pipelineInformer pipelineinformer.TektonPipelineInformer
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ tektonchainsreconciler.Interface = (*Reconciler)(nil)
+var _ tektonchainsreconciler.Finalizer = (*Reconciler)(nil)
+
+const createdByValue = "TektonChains"
+
+var (
+	ls = metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.CreatedByKey:     createdByValue,
+			v1alpha1.InstallerSetType: v1alpha1.ChainsResourceName,
+		},
+	}
+)
+
+// ReconcileKind compares the actual state with the desired, and attempts to
+// converge the two.
+func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonChains) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+	tc.Status.InitializeConditions()
+	tc.Status.ObservedGeneration = tc.Generation
+
+	logger.Infow("Reconciling TektonChains", "status", tc.Status)
+
+	if tc.GetName() != v1alpha1.ChainsResourceName {
+		msg := fmt.Sprintf("Resource ignored, Expected Name: %s, Got Name: %s",
+			v1alpha1.ChainsResourceName,
+			tc.GetName(),
+		)
+		logger.Error(msg)
+		tc.Status.MarkNotReady(msg)
+		return nil
+	}
+
+	// find a valid TektonPipeline installation
+	if _, err := common.PipelineReady(r.pipelineInformer); err != nil {
+		if err.Error() == common.PipelineNotReady {
+			tc.Status.MarkDependencyInstalling("TektonPipeline is still installing")
+			// wait for TektonPipeline status to change
+			return fmt.Errorf(common.PipelineNotReady)
+		}
+		// (tektonpipeline.operator.tekton.dev instance not available yet)
+		tc.Status.MarkDependencyMissing("TektonPipeline does not exist")
+		return err
+	}
+	tc.Status.MarkDependenciesInstalled()
+
+	// Pass the object through defaulting
+	tc.SetDefaults(ctx)
+
+	// Mark TektonChains Instance as Not Ready if an upgrade is needed
+	if err := r.markUpgrade(ctx, tc); err != nil {
+		return err
+	}
+
+	if err := tektoninstallerset.CleanUpObsoleteResources(ctx, r.operatorClientSet, createdByValue); err != nil {
+		return err
+	}
```

Comment:
this wouldn't be required for chains.
as chains have the new installation flow from start

Reply:
removed